static/js/hotkey.js

```diff
@@ -357,6 +358,10 @@ function process_hotkey(e) {
         case 'narrow_by_subject':
             return do_narrow_action(narrow.by_subject);
         case  'enter': // Enter: respond to message (unless we need to do something else)
+            if (current_view === "#settings" || current_view === "#administration" ||
+                current_view === "#subscriptions") {
+                return false;
+            }
```

Comment:
@timabbott I tried making the recommended changes but it didn't resolve the bug.
I believe this reason for this is that we need to keep a check where the shortcuts to operating a message are in place.
The switch case around line 354-355 in `hotkey.js` is responsible for handling the shortcuts that cater to the message box. This was the reason I placed the condition over there and it worked out.

Please let me know further improvements, in any. 

Thanks!